### PR TITLE
Document release procedure (Lombiq Technologies: OCORE-22)

### PR DESCRIPTION
**Pull request by Piedone**
For https://github.com/OrchardCMS/OrchardCore/issues/5841.

### CODE REVIEW

File: src/docs/topics/publishing-releases/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```
+### Prepare the project
+Do some housekeeping on GitHub in the [main repo](https://github.com/OrchardCMS/OrchardCore).
+
+- [ ] Close remaining issues for the version (including merging corresponding pull requests if suitable) or assign then to the next one.
+- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).
+
+### Prepare the code
+Update the source so everything looks like on the new version.
```
Comment (agriffard) [inline src/docs/topics/publishing-releases/README.md:21]: "everything looks like on the new version" Is there a word missing?

File: mkdocs.yml [hunk 2]
```diff
@@ -88,6 +88,7 @@ nav:
 #        - Data: docs/topics/data/README.md
 #        - Configuration: docs/topics/configuration/README.md
         - Workflows: docs/topics/workflows/README.md
+		- Publishing a new release: docs/topics/publishing-releases/README.md
```
Comment (agriffard) [inline mkdocs.yml:91]: The number of spaces is important. It is a yaml file.

File: src/docs/topics/publishing-releases/README.md (hunk 1 above)
Comment (Piedone) [reply src/docs/topics/publishing-releases/README.md:21]: No, what I want to say that change the code until it looks like what you want to have for the new version.

File: mkdocs.yml (hunk 2 above)
Comment (Piedone) [reply mkdocs.yml]: Yeah, it was actually tabs my mistake.

**Comment by agriffard** [comment]
Add comparison link between 2 releases. Ex:
https://github.com/orchardcms/orchardcore/compare/1.0.0-beta3...1.0.0-rc1

**Comment by sebastienros** [comment]
Why can't I edit the files in this PR? Have you removed the permissions for that? I have experienced that many times recently, maybe GH changed this?

**Comment by Piedone** [comment]
Hmm, not at all, didn't removed any permissions. Probably because it was a draft PR? Changed it now.

**Comment by Piedone** [comment]
Nothing to add here anymore.

**Comment by agriffard** [comment]
Let's also add a description on the GitHub release.

https://github.com/OrchardCMS/OrchardCore/releases/tag/1.0.0-rc2 is empty right now.

Do we just paste the list of PRs merged?

**Comment by Piedone** [comment]
Oops, removed misleading label.

I think let's keep release notes in one place, so rather link to it. I.e.

```
For details on this version see the [release notes in the documentation](https://docs.orchardcore.net/en/dev/docs/releases/1.0.0-rc2/).
```

I've also added this to the docs.

**Comment by Piedone** [comment]
Anybody with any other suggestions or can we merge? I've updated the release's description.

**Comment by agriffard** [comment]
We should create a release of the translations: https://github.com/OrchardCMS/OrchardCore.Translations/releases

We didn't do it for RC1 and RC2.

**Comment by Piedone** [comment]
Wouldn't one just download these from Crowdin directly for a nightly or use one of the packages like https://www.nuget.org/packages/OrchardCore.Translations.fr/1.0.0-rc2-108003 for a release?

**Comment by agriffard** [comment]
I think we can merge this process @Piedone

**Comment by Piedone** [comment]
Sounds good!

**Pull request merged**